File: applications/v2/frontend/src/app/development/models/person.ts
import { PageInfo } from './page';

export type PersonStatus = 'ACTIVE' | 'INACTIVE';

export interface Person {
  id: string;
  name: string;
  email: string;
  role: string;
  status: PersonStatus;
}

export interface PersonSummary {
  id: string;
  name: string;
}

export interface PersonCreateRequest {
  name: string;
  email: string;
  role: string;
  status: PersonStatus;
}

export interface PersonUpdateRequest {
  name: string;
  email: string;
  role: string;
  status: PersonStatus;
}

export interface PeoplePage {
  items: Person[];
  page: PageInfo;
}

File: applications/v6/backend/src/main/java/com/example/weeklyplanning/service/dto/Person.java
package com.example.weeklyplanning.service.dto;

import com.example.weeklyplanning.domain.AccountRole;
import com.example.weeklyplanning.domain.PersonStatus;

import java.time.Instant;
import java.util.UUID;

public record Person(
    UUID id,
    UUID accountId,
    String name,
    String username,
    String email,
    AccountRole role,
    PersonStatus status,
    Instant createdAt,
    Instant updatedAt
) {
}
